exporter/exportertest/contract_checker.go

```diff
@@ -0,0 +1,260 @@
+	fmt.Printf("Number of non-permanent errors: %d\n", reqCounter.error.nonpermanent)
+	checkIfTestPassed(params.T, params.NumberOfTestElements, reqCounter)
+}
+
+func checkTraces(params CheckConsumeContractParams, mockReceiver MockReceiver, checkIfTestPassed func(*testing.T, int, RequestCounter)) {
+	ctx := context.Background()
+
+	var exp exporter.Traces
+	var err error
+	exp, err = params.Factory.CreateTracesExporter(ctx, NewNopCreateSettings(), params.Config)
+	require.NoError(params.T, err)
+	require.NotNil(params.T, exp)
+
+	err = exp.Start(ctx, componenttest.NewNopHost())
+	require.NoError(params.T, err)
+
+	defer func(exp exporter.Traces, ctx context.Context) {
+		err = exp.Shutdown(ctx)
+		require.NoError(params.T, err)
+		err = mockReceiver.Stop()
+		require.NoError(params.T, err)
+	}(exp, ctx)
+
+	for i := 0; i < params.NumberOfTestElements; i++ {
+		id := UniqueIDAttrVal(strconv.Itoa(i))
+		fmt.Println("Preparing trace number: ", id)
+		data := CreateOneTraceWithID(id)
+
+		err = exp.ConsumeTraces(ctx, data)
+	}
+
+	reqCounter := mockReceiver.RequestCounter()
+	// The overall number of requests sent by exporter
+	fmt.Printf("Number of export tries: %d\n", reqCounter.total)
+	// Successfully delivered items
+	fmt.Printf("Total items received successfully: %d\n", reqCounter.success)
+	// Number of errors that happened
+	fmt.Printf("Number of permanent errors: %d\n", reqCounter.error.permanent)
+	fmt.Printf("Number of non-permanent errors: %d\n", reqCounter.error.nonpermanent)
+	checkIfTestPassed(params.T, params.NumberOfTestElements, reqCounter)
+}
+
+func checkLogs(params CheckConsumeContractParams, mockReceiver MockReceiver, checkIfTestPassed func(*testing.T, int, RequestCounter)) {
+	ctx := context.Background()
+
+	var exp exporter.Logs
+	var err error
+	exp, err = params.Factory.CreateLogsExporter(ctx, NewNopCreateSettings(), params.Config)
+	require.NoError(params.T, err)
+	require.NotNil(params.T, exp)
+
+	err = exp.Start(ctx, componenttest.NewNopHost())
+	require.NoError(params.T, err)
+
+	defer func(exp exporter.Logs, ctx context.Context) {
+		err = exp.Shutdown(ctx)
+		require.NoError(params.T, err)
+		err = mockReceiver.Stop()
+		require.NoError(params.T, err)
+	}(exp, ctx)
+
+	for i := 0; i < params.NumberOfTestElements; i++ {
+		id := UniqueIDAttrVal(strconv.Itoa(i))
+		fmt.Println("Preparing log number: ", id)
+		data := CreateOneLogWithID(id)
+
+		err = exp.ConsumeLogs(ctx, data)
+	}
+
+	reqCounter := mockReceiver.RequestCounter()
+	// The overall number of requests sent by exporter
+	fmt.Printf("Number of export tries: %d\n", reqCounter.total)
+	// Successfully delivered items
+	fmt.Printf("Total items received successfully: %d\n", reqCounter.success)
+	// Number of errors that happened
+	fmt.Printf("Number of permanent errors: %d\n", reqCounter.error.permanent)
+	fmt.Printf("Number of non-permanent errors: %d\n", reqCounter.error.nonpermanent)
+	checkIfTestPassed(params.T, params.NumberOfTestElements, reqCounter)
```

Comment:
We cannot check this synchronously. The test should pass with the enabled exporter's queue as well. Please use `assert.Eventually` instead

Reply:
Done with `assert.EventuallyWithT`, so I didn't need to rewrite functions to return bool